huaweicloud/services/dds/resource_huaweicloud_dds_lts_log.go

```diff
@@ -0,0 +1,338 @@
+	}
+
+	return resourceDdsLtsLogRead(ctx, d, meta)
+}
+
+func resourceDdsLtsLogRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var mErr *multierror.Error
+	var (
+		getDdsLtsLogHttpUrl = "v3/{project_id}/instances/logs/lts-configs"
+		getDdsLtsLogProduct = "dds"
+	)
+
+	getDdsLtsLogClient, err := cfg.NewServiceClient(getDdsLtsLogProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DDS Client: %s", err)
+	}
+
+	getDdsLtsLogPath := getDdsLtsLogClient.Endpoint + getDdsLtsLogHttpUrl
+	getDdsLtsLogPath = strings.ReplaceAll(getDdsLtsLogPath, "{project_id}", getDdsLtsLogClient.ProjectID)
+
+	getDdsLtsLogResp, err := pagination.ListAllItems(
+		getDdsLtsLogClient,
+		"offset",
+		getDdsLtsLogPath,
+		&pagination.QueryOpts{MarkerField: ""})
+	if err != nil {
+		return diag.Errorf("error retrieving DDS LTS configs: %s", err)
+	}
+
+	getDdsLtsLogRespJson, err := json.Marshal(getDdsLtsLogResp)
+	if err != nil {
+		return diag.Errorf("error marshaling DDS LTS configs: %s", err)
+	}
+
+	var getDdsLtsLogRespBody interface{}
+	err = json.Unmarshal(getDdsLtsLogRespJson, &getDdsLtsLogRespBody)
+	if err != nil {
+		return diag.Errorf("error unmarshaling DDS LTS configs: %s", err)
+	}
+
+	instanceLtsConfig := FilterInstanceLtsConfig(d.Id(), getDdsLtsLogRespBody)
+	instance := utils.PathSearch("instance", instanceLtsConfig, nil)
+	ltsConfigs := utils.PathSearch("lts_configs", instanceLtsConfig, make([]interface{}, 0)).([]interface{})
+	if len(ltsConfigs) == 0 || !utils.PathSearch("enabled", ltsConfigs[0], false).(bool) {
+		return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "")
+	}
+
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("instance_id", utils.PathSearch("id", instance, nil)),
+		d.Set("log_type", utils.PathSearch("log_type", ltsConfigs[0], nil)),
+		d.Set("lts_group_id", utils.PathSearch("lts_group_id", ltsConfigs[0], nil)),
+		d.Set("lts_stream_id", utils.PathSearch("lts_stream_id", ltsConfigs[0], nil)),
+		d.Set("enabled", utils.PathSearch("enabled", ltsConfigs[0], nil)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func buildDeleteDdsLtsLogBodyParams(d *schema.ResourceData) map[string]interface{} {
+	ltsConfigs := map[string]interface{}{
+		"lts_configs": []map[string]interface{}{
+			{
+				"instance_id": d.Get("instance_id"),
+				"log_type":    d.Get("log_type"),
+			},
+		},
+	}
+	return ltsConfigs
+}
+
+func resourceDdsLtsLogDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var (
+		deleteDdsLtsLogHttpUrl = "v3/{project_id}/instances/logs/lts-configs"
+		deleteDdsLtsLogProduct = "dds"
+	)
+	deleteDdsLtsLogClient, err := cfg.NewServiceClient(deleteDdsLtsLogProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DDS Client: %s", err)
```

Comment:
`client` should be in lowercase, please fix others
```suggestion
		return diag.Errorf("error creating DDS client: %s", err)
```

Reply:
done